lib/src/react16_suggestors/react_style_maps_updater.dart

```diff
@@ -0,0 +1,299 @@
+    with AstVisitingSuggestorMixin
+    implements Suggestor {
+  @override
+  visitCascadeExpression(CascadeExpression node) {
+    super.visitCascadeExpression(node);
+
+    for (Expression cascade in node.cascadeSections) {
+      if (!hasValidationComment(node, sourceFile)) {
+        if (cascade.toString().contains('style') &&
+            !cascade.toString().contains('setProperty')) {
+          /// A style map, method invocation, or a variable
+          dynamic stylesObject = getStyles(cascade);
+
+          /// CSS properties that were modified by the script or need to be
+          /// manually checked
+          List<String> affectedValues = [];
+
+          // Variables that affect the writing of the comment that goes above
+          // the style prop. There are different messages patched based on
+          // these booleans.
+          bool styleMapContainsAVariable = false;
+          bool isAVariable = false;
+          bool isAFunction = false;
+          bool isForCustomProps =
+              cascade.toString().toLowerCase().contains('props');
+
+          if (stylesObject is MapLiteral) {
+            stylesObject.entries.forEach((MapLiteralEntry cssPropertyRow) {
+              String originalCssPropertyKey =
+                  cssPropertyRow.beginToken.toString();
+              String cleanedCssPropertyKey =
+                  cleanString(originalCssPropertyKey);
+
+              // We cannot simply do `cssPropertyRow.endToken` because if the
+              // "end token" is an expression, it only returns the last
+              // value in the ternary. Removing the beginning token keeps the
+              // ternary intact.
+              List endToken = cssPropertyRow.childEntities.toList();
+              endToken.removeRange(0, 2);
+              String originalCssPropertyValue = endToken.join(" ");
+              String cleanedCssPropertyValue =
+                  cleanString(originalCssPropertyValue);
+
+              bool rowContainsAVariable =
+                  nodeIsLikelyAVariable(originalCssPropertyValue);
+
+              num end = cssPropertyRow.end;
+
+              // If the codemod does not override the commas originally in
+              // the code, the formatting can be undesirable. As a result,
+              // the codemod manually places a comma after a modified
+              // property. To do this, the script needs to override the
+              // already existing comma (one character after the row end)
+              // without overriding the closing bracket (cascade.end).
+              if (end + 1 != cascade.end) {
+                end += 1;
+              }
+
+              if (rowContainsAVariable) {
+                affectedValues.add(cleanedCssPropertyKey);
+
+                if (!styleMapContainsAVariable) {
+                  styleMapContainsAVariable = true;
+                }
+              }
+
+              if (nodeIsLikelyAnExpression(originalCssPropertyValue)) {
```

Comment:
It would be a bit simpler and more robust to use the AST for this:
```dart
final value = cssPropertyRow.value;
if (value is ConditionalExpression) { // ConditionalExpression is the ternary AST node
  final ternaryExpression1 = value.then;
  final ternaryExpression2 = value.otherwise;
}
```

Reply:
Nice, that is better!